src/test/storage/dictionary_segment_test.cpp

```diff
@@ -7,51 +7,97 @@
+};
+
+TEST_F(StorageDictionarySegmentTest, CompressSegmentString) {
+  vc_str->append("Bill");
+  vc_str->append("Steve");
+  vc_str->append("Alexander");
+  vc_str->append("Steve");
+  vc_str->append("Hasso");
+  vc_str->append("Bill");
+
+  auto col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("string", vc_str);
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<std::string>>(col);
+
+  // Test attribute_vector size
+  EXPECT_EQ(dict_col->size(), 6u);
+
+  // Test dictionary size (uniqueness)
+  EXPECT_EQ(dict_col->unique_values_count(), 4u);
+
+  // Test sorting
+  auto dict = dict_col->dictionary();
+  EXPECT_EQ((*dict)[0], "Alexander");
+  EXPECT_EQ((*dict)[1], "Bill");
+  EXPECT_EQ((*dict)[2], "Hasso");
+  EXPECT_EQ((*dict)[3], "Steve");
+
+  // Test memory usage estimation (based on very short strings)
+  auto attribute_vector_size = 6u * sizeof(uint8_t);
+  auto dictionary_size = 4u * sizeof(std::string);
+  EXPECT_EQ(dict_col->estimate_memory_usage(), attribute_vector_size + dictionary_size);
+}
+
+TEST_F(StorageDictionarySegmentTest, LowerUpperBound) {
+  for (int i = 0; i <= 10; i += 2) vc_int->append(i);
+  auto col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("int", vc_int);
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+
+  EXPECT_EQ(dict_col->lower_bound(4), (opossum::ValueID)2);
+  EXPECT_EQ(dict_col->upper_bound(4), (opossum::ValueID)3);
+
+  EXPECT_EQ(dict_col->lower_bound(5), (opossum::ValueID)3);
+  EXPECT_EQ(dict_col->upper_bound(5), (opossum::ValueID)3);
+
+  EXPECT_EQ(dict_col->lower_bound(15), opossum::INVALID_VALUE_ID);
+  EXPECT_EQ(dict_col->upper_bound(15), opossum::INVALID_VALUE_ID);
+}
+
+TEST_F(StorageDictionarySegmentTest, CorrectAttributeVectorWidth) {
+  for (int i = 0; i <= 10; i += 2) vc_int->append(i);
+  auto col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("int", vc_int);
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+
+  EXPECT_EQ(dict_col->attribute_vector()->width(), sizeof(uint8_t));
+
+  // append 2 ^ 8 more values
+  for (int i = 0; i <= 256; i++) vc_int->append(i);
+
+  col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("int", vc_int);
+  dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+  EXPECT_EQ(dict_col->attribute_vector()->width(), sizeof(uint16_t));
+
+  /*  takes too long
+  // append 2 ^ 16 more values
+  for (int i = 0; i <= 65536; i++) vc_int->append(i);
+
+  col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("int", vc_int);
+  dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+  EXPECT_EQ(dict_col->attribute_vector()->width(), sizeof(uint32_t));
+  */
+}
```

Comment:
who is this code for?

Reply:
For you to see that we considered this case ;) I will remove this.